app/controllers/api/v1/schedules_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::V1::SchedulesController < ApplicationController
+
+  def show 
+    schedule = Schedule.find_by(user_id: params[:user_id])
+    if schedule 
+      render json: ScheduleSerializer.new(schedule).serializable_hash.to_json
+    else
+      render json: { error: "Schedule not found" }, status: :not_found
+    end
+  end
+
+  def update
+    schedule = Schedule.find_by(user_id: params[:user_id])
+    show = Show.find_by(id: params[:show_id])
+
+    if schedule && show
+      if params[:action_type] == 'add'
+        if schedule.shows.count < 8
+          schedule.shows << show unless schedule.shows.include?(show)
```

Comment:
Maybe look into the ruby [set](https://ruby-doc.org/stdlib-2.7.3/libdoc/set/rdoc/Set.html) data structure here.